Open-Rival/src/UnitAnimation.cpp

```diff
@@ -45,7 +50,14 @@ namespace Rival {
     }
 
     void UnitAnimation::tick() {
-        auto spritesheetSpan = spritesheetEntry.second - spritesheetEntry.first + 1;
-        animationStep = (animationStep + 1) % (spritesheetSpan);
+        auto spritesheetSpan = std::get<1>(spritesheetEntry) - std::get<0>(spritesheetEntry) + 1;
+        if (1 == spritesheetSpan) {
+            return;
+        }
+
+        while (getCurrentTime() > lastAnimationUpdate + std::get<2>(spritesheetEntry)) {
```

Comment:
If a spritesheetEntry ever has an animation time of 0, this will cause an infinite loop.

Reply:
The way you have written this, animations will be tied to real time. This is not too much of a problem, but it means that if a lag spike occurs, the animation will "jump" to catch up. This might look strange in a fixed-timestep game, because movement and other events are *not* tied to real time - they are tied to the tick rate of the logic.

I would suggest that we do the same for animations, that is, rather than check how much real time has passed, just pretend every frame that X ms have passed, where X is the timestep of the logic (16ms). This should keep everything nicely in-sync.

Reply:
A spritesheetEntry with more than 1 spreadsheetSpan has animation time of 0, then the loop will be infinite freezing the game, yes.

Animations are a visual feedback to a human who has a sense of real-time, not ticks. we can discuss this on Discord since I am not getting this particular concern.